src/main/java/hillel/spring/appointments/Appointment.java

```diff
@@ -19,6 +16,8 @@
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Integer id;
+    @Version
+    private Integer version;
```

Comment:
I do not see how this lock could work.
The only way to make it work - is to save **stale** object to DB. To do that you should read-modify-save it. 
But in your code, I only see just save.
So here are two ways:
* modify your model such a way, to use read-modify-save pattern on creating an appointment
* find the other way to make sure that there are no duplicates in DB